src/alchemlyb/tests/test_preprocessing.py

```diff
@@ -213,6 +236,97 @@ def test_raise_ValueError_for_mismatched_data(self, series):
         with pytest.raises(ValueError):
             self.slicer(data, series=series)
 
+    @pytest.mark.parametrize(('data', 'lower', 'upper'),
+                             [
+                                 (gmx_benzene_dHdl(), 1000, 34000),
+                                 (gmx_benzene_u_nk(), 1000, 34000),
+                             ])
+    def test_lower_and_upper_bound_slicer(self, data, lower, upper):
+        """
+        Test that the lower and upper time is respected when using statistical_inefficiency
+        without a series. In this case, statistical_inefficiency should behave like slicing
+        """
+        original_length = len(data)
+        # Check that the input data is appropriate for the test
+        assert any(data.reset_index()['time'] < lower)
+        assert any(data.reset_index()['time'] > upper)
```

Comment:
I am not 100% convinced that this is necessary, given that you are the one choosing the test data. It guards against the case when someone else adds other data to the parameters, so I am not totally opposed to having it here and elsewhere. However, in order to reduce code duplication, can you make it a small function such as `_assert_data_has_boundaries(data, lowe, upper)` or rather choose a better name.

Reply:
I just saw your [explanation for why you include these guards everywhere](https://github.com/alchemistry/alchemlyb/pull/199/files#r909367088). I am ok with leaving it, I'd just reduce copy&paste code.

Reply:
Done in latest version.